documentation/src/test/java/example/exception/ExceptionAssertionDemo.java

```diff
@@ -18,15 +19,23 @@
 // tag::user_guide[]
 
 class ExceptionAssertionDemo {
+
     @Test
     void testExpectedExceptionIsThrown() {
-        assertThrows(IllegalArgumentException.class, () -> {
+        // The following assertion succeeds because the code under assertion throws
+        // the expected IllegalArgumentException
+        // The assertion also returns the thrown exception which can be used for
+        // further assertions like asserting the exception messages
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> {
             throw new IllegalArgumentException("Expected exception");
-        }, "Expected IllegalArgumentException to be thrown");
+        });
+        assertEquals(exception.getMessage(), "Expected exception");
```

Comment:
The arguments should be switched; these methods define left as _expected_ and right as _actual_, see the [`assertEquals` documentation](https://junit.org/junit5/docs/current/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertEquals(java.lang.Object,java.lang.Object)).

Otherwise the assertion error thrown on mismatch will be quite confusing.
```suggestion
        assertEquals("Expected exception", exception.getMessage());
```